### ISSUE TITLE
PEP 596 compatibility (Python3.9)

### ISSUE THREAD

**Issue by Zeitsperre** [Ouranosinc/xclim#566]
The release version of Python3.9 is expected one week from today (https://www.python.org/dev/peps/pep-0596/)

Modifying our existing code for compatibility and integrating python3.9 into the Travis-CI build tests will require some effort. Given that compatibility with earlier Python versions must be maintained, we won't be able to use new features (until Python3.6 is deprecated in 2021). For those interested, here are the new goodies:

    PEP 584, Add Union Operators To dict
    PEP 585, Type Hinting Generics In Standard Collections
    PEP 593, Flexible function and variable annotations
    PEP 614, Relaxing Grammar Restrictions On Decorators
    PEP 615, Support for the IANA Time Zone Database in the Standard Library
    PEP 616, String methods to remove prefixes and suffixes
    PEP 617, New PEG parser for CPython

**Comment by huard** [comment]
I'd set a low priority for this. We could check that nothing breaks with 3.9, but I'd defer time-consuming changes until there is a user base for it.

**Comment by Zeitsperre** [comment]
We'll need to wait on builds for a bunch of dependencies, just like we had to for Python3.8. Once `xarray` starts supporting Python3.9, it would be good to follow suit.

**Comment by aulemahal** [comment]
Quick heads up, I installed a new python3.9 environment on my machine and ran all xclim tests. Everything runs fine except the loess methods on xclim.sdba, which fail with numba errors. I'm not looking into it for now, but it could only be because of a recent numba version. So AFAIK, Python3.9 shouldn't break xclim!

**Comment by Zeitsperre** [comment]
I don't think it will either, so long as we are maintaining backwards compatibility to Python3.6, the only work involved here would be setting up a build for it in `.travis.yml`.
